### PR TITLE
Initialize all shader caches concurrently

### PR DESCRIPTION
**Pull request by marco-carvalho**

### CODE REVIEW

File: src/Ryujinx.Graphics.Gpu/GpuContext.cs [hunk 1]
```diff
@@ -274,14 +276,11 @@ private void ShaderCacheStateUpdate(ShaderCacheState state, int current, int tot
...
-        public void InitializeShaderCache(CancellationToken cancellationToken)
+        public async Task InitializeShaderCache(CancellationToken cancellationToken)
         {
             HostInitalized.WaitOne();
 
-            foreach (var physicalMemory in PhysicalMemoryRegistry.Values)
-            {
-                physicalMemory.ShaderCache.Initialize(cancellationToken);
-            }
+            await Task.WhenAll(PhysicalMemoryRegistry.Values.Select(pm => pm.ShaderCache.Initialize(cancellationToken)));
```
Comment (mattzink) [inline src/Ryujinx.Graphics.Gpu/GpuContext.cs:283]: Shouldn't `Initialize()` be renamed to `InitializeAsync()`?

**Comment by tgcordell** [comment]
What’s the expected behavior change for the end user for this? Some type of performance improvement, but where? During initial game load?

**Comment by marco-carvalho** [comment]
In the initial load of compiling shaders

**Comment by Foul-Tarnished** [comment]
Each game has only a single shader cache though ?
Did you ask copilot to make ryujinx faster or what lmao

> The change should make shader cache initialization happen concurrently instead of one by one. This means the game should start up faster because it takes less time to get everything ready at the beginning. So, users should notice a quicker load time when they first launch the game.

**Pull request closed without merging**